Prepare EntryConvert to allow localized possible values

jsonBackup opened the pull request:
Closes #400
Related https://github.com/PHOENIXCONTACT/ngx-moryx-web/pull/28

1nf0rmagician commented on src/Moryx/Serialization/EntryConvert/EntryToModel/EntryCollectionBase.cs:
```diff
@@ -52,7 +52,7 @@ public string CurrentType
         /// Possible entries that can be used to create instances
         /// </summary>
-        public string[] PossibleTypes => CollectionRoot.Value.Possible;
+        public string[] PossibleTypes => CollectionRoot.Value.Possible?.Select(p => p?.Key).ToArray();
```
Can you also make this an object array instead of string, classes could also use `DisplayName` and `Description 🤔`

1nf0rmagician commented on src/Moryx/Serialization/EntryConvert/EntryConvert.cs:
```diff
@@ -495,6 +495,26 @@ private static Entry ConvertParameter(ParameterInfo parameter, ICustomSerializat
+        /// Convert string[] possible values to EntryPossible[].
+        /// Key and DisplayName are set to the string value.
+        /// </summary>
+        private static EntryPossible[] ConvertPossible(string[] possible)
```
I think we should move the convertion one level up already, i.e. the API of the `serialization.PossibleValues(parameterType, parameter)` should already return the `EntryPossible`, don't you think? 🤔

dbeuchler replied:
I also think so, `ICustomSerialization` should return `EntryPossible`

dbeuchler commented on src/Moryx/Serialization/EntryConvert/EntryPossible.cs:
```diff
@@ -0,0 +1,29 @@
+    /// Possible value for an entry.
+    /// </summary>
+    [DataContract]
+    public class EntryPossible
```
I also thought about that, but in this case `EntryPossible` is okay for me to be consistent with the other `Entry` related classes

1nf0rmagician commented on src/Moryx/Configuration/PossibleValuesSerialization.cs:
```diff
@@ -93,7 +93,7 @@ public override string[] PossibleElementValues(Type memberType, ICustomAttribute
 
             // Use attribute
             var values = valuesAttribute.GetValues(Container, ServiceProvider);
```
Do we need to enhance the API of `GetValues` on the attribute or the Attribute itself as well?

dbeuchler reviewed (changes requested):
Looks good so far. What about 
- all UI updates, every usage of the entry editor should be broken?
- Migration guide information

jsonBackup commented:
> Looks good so far. What about
> 
>     * all UI updates, every usage of the entry editor should be broken?
> 
>     * Migration guide information

I've implemented the required changes in Moryx.CommandCenter.Web so the entry editor usages are now aligned with the new `EntryPossible` handling. Since https://github.com/PHOENIXCONTACT/MORYX-Framework/pull/792 no other web projects reference EntryValue directly anymore, so they should not be broken by this but require the new version of ngx-moryx-web with the changes I've implemented here: https://github.com/PHOENIXCONTACT/ngx-moryx-web/pull/28.
Because of the ongoing supply-chain incident, I'm currently holding off on updating the package.json files to reference the new ngx-moryx-web version.

I've also extended the migration guide :)
